### ISSUE TITLE
Weird source files in multilevel CMake build

### ISSUE THREAD

**Issue by TimPushkin** [scapix-com/scapix#38]
I have a CMake project with layout like this:
```
project_root/
    examples/
        CMakeLists.txt
        <some executable examples>
    external/
        CMakeLists.txt
        scapix/
                CMakeLists.txt  <-- Scapix is loaded here
    include/
        <headers to parse are here>
    src/
        CMakeLists.txt  <-- The main library target is created here
        <sources are here>
    CMakeLists.txt
```

The top-level CMake file includes other files via `add_subdirectory`: firstly the main target is created in `src` subfolder via `add_library`, then Scapix and other libraries are configured, then `examples` subfolder is added the same way, containing some executable targets created via `add_executable` (the library is linked to them). The project builds and works fine without Scapix.

The problem is that when I add Scapix to the project with
```
scapix_bridge_headers(<main library target> com.example.something <path to SomeClass.h>)
```
and run CMake with `-DSCAPIX_BRIDGE=java` I get the following errors at configure time:
```
CMake Error at src/CMakeLists.txt:3 (add_library):
  Cannot find source file:

    <my project path>/generated/bridge/java/SomeClass.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc


CMake Error at examples/CMakeLists.txt:2 (add_executable):
  Cannot find source file:

    <my project path>/generated/bridge/java/com/example/something/SomeClass.java

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
```

I do not specify the missing files by myself, it seems like Scapix adds them to the targets somehow.

I can get rid of the first one if I place the content of the CMake file that loads Scapix into the CMake file that creates the library target (`src/CMakeLists.txt`), but then the second one remains, and I can get rid of the second one if I place it into the CMake file that creates the example target (`examples/CMakeLists.txt`), but then the first one remains.

Is it possible to fix these errors without throwing all CMake code into a single CMake file?

**Comment by Boris-Rasin** [comment]
Thank you for reporting this.

Is your project available on github?

If not, can you please post skeleton CMake files, so I can setup the test project?

**Comment by TimPushkin** [comment]
Thanks for the response and sorry for the wait time.

I'm actually having troubles trying to reproduce the error on this smaller scale project. The project I'm having problems with is available [here](https://github.com/TimPushkin/SgbmAndroid/tree/054e6e17b054d77f5704b045b3320510cb22b807/sgbmandroid/src/main/cpp) -- it is a subproject of an Android library, but it is configured to build fine without Android dependencies.

While trying to reproduce the error, I noticed that it goes away if I set `cmake_minimum_required` to 3.20 or higher (currently it is set to 3.18 as it is the latest version prepackaged with Android Studio).

**Comment by TimPushkin** [comment]
I managed to recreate the error and commited it to the repository. It represents the current state of my project: if I use CMake 3.18 then the errors described above appear, if I use CMake 3.20 or higher then the configuration completes without errors, but build fails because the bridge file for `SomeClass.cpp` doesn't get generated and thus cannot be found.

**Comment by Boris-Rasin** [comment]
Turns out there were multiple issues affecting the situation when `add_library` and `scapix_bridge_headers` are called from different directories.

I submitted a fix, and now it should work correctly with CMake 3.18 or later (there is still an issue with CMake older then 3.18).

All you need in your case is a single call to `scapix_fix_sources(lib)` in every CMake file where `lib` is used, even if it's used multiple times in the same CMake file (but there is no harm in calling it multiple times). It should work now even if all these CMake files are skipped.

I will release new Scapix version soon.

**Comment by Boris-Rasin** [comment]
Fixed in cmodule v1.0.34:

https://github.com/scapix-com/cmodule/releases

**Comment by TimPushkin** [comment]
Can confirm that I no longer have these configure time errors with v1.0.34, thanks
